### PR TITLE
Lint clean ups

### PR DESCRIPTION
**Pull request by NilsGraf**
These changes are cleaning up some lint issues we are seeing when running Ascentlint from Real Intent. Let me know what you think. Thanks!

### CODE REVIEW

File: src/dm_csrs.sv [hunk 1]
```diff
@@ -218,7 +218,7 @@ module dm_csrs #(
         progbuf_d   = progbuf_q;
         data_d      = data_q;
         sbcs_d      = sbcs_q;
-        sbaddr_d    = sbaddress_i;
```
Comment (zarubaf) [inline src/dm_csrs.sv]: I am in favor of fixing this properly tbh:

1. Make `sbaddr_[d|q]` width a parameter of `BusWidth`. 
2. Disabling (e.g. returning an error) if the debugger accesses `dm::SBAddress1` in the case `BusWidth` is `32`

File: src/dmi_jtag.sv [hunk 2]
```diff
@@ -137,7 +137,7 @@ module dmi_jtag #(
                 end
             end
 
-            WaitWriteValid: begin
+            default: begin
```
Comment (zarubaf) [inline src/dmi_jtag.sv]: I'd suggest a blank `default` item. Removing this valid state makes the FSM harder to read.

File: src/dm_pkg.sv [hunk 3]
```diff
@@ -25,9 +25,9 @@ package dm;
     localparam logic [3:0] DataCount     = 4'h2;
 
     // address to which a hart should jump when it was requested to halt
-    localparam logic [63:0] HaltAddress = 64'h800;
-    localparam logic [63:0] ResumeAddress = HaltAddress + 4;
-    localparam logic [63:0] ExceptionAddress = HaltAddress + 8;
+    localparam logic [31:0] HaltAddress = 32'h800;
+    localparam logic [31:0] ResumeAddress = HaltAddress + 4;
+    localparam logic [31:0] ExceptionAddress = HaltAddress + 8;
```
Comment (zarubaf) [inline src/dm_pkg.sv]: We will need to split this into `32` and `64` bit variants. As this change now works for the 32 bit case but not for the 64.

File: src/dmi_jtag.sv (hunk 2 above)
Comment (NilsGraf) [reply src/dmi_jtag.sv]: Sure I can do this. Actually, the linting tool says that state "WaitWriteValid" can never be reached. And looking at the code, I tend to agree with this. Is that expected? Is that perhaps a bug or can we remove WaitWriteValid altogether?

File: src/dm_pkg.sv (hunk 3 above)
Comment (NilsGraf) [reply src/dm_pkg.sv]: Ok, I can revert this change and then simply waive this lint error.

Alternatively, do you want to define a parameter AddrWidth, and change it as follows?
localparam logic [AddrWidth-1:0] HaltAddress = AddrWidth'(12'h800);
localparam logic [AddrWidth-1:0] ResumeAddress = HaltAddress + 4;
localparam logic [AddrWidth-1:0] ExceptionAddress = HaltAddress + 8;

File: src/dm_csrs.sv (hunk 1 above)
Comment (NilsGraf) [reply src/dm_csrs.sv]: Ok, this looks like a major change for the design. For now, the bit-width casting to 64 bits works well for both 64 and 32 bit versions. Alternatively, I can remove the bitwidth-casting and write a lint-waiver instead.

File: src/dm_pkg.sv (hunk 3 above)
Comment (bluewww) [reply src/dm_pkg.sv]: AddrWidth would be BusWidth which is a parameter in `dm_top`. I don't know a way to share this setting except for using macros. I think revert is the best.

File: src/dm_csrs.sv (hunk 1 above)
Comment (bluewww) [reply src/dm_csrs.sv]: The spec says (according to my interpretation) accessing `dm::SBAddress` when `sbasize < 33 (sbasize=BusWidth)` then we read 0

Comment (NilsGraf) [reply src/dm_csrs.sv]: I reverted this change.

File: src/dm_pkg.sv (hunk 3 above)
Comment (NilsGraf) [reply src/dm_pkg.sv]: I reverted this change.

**Comment by NilsGraf** [comment]
Hi @zarubaf and @bluewww , I reverted all controversial changes. Please take another look and let me know if the other changes look good to you.

File: src/dm_csrs.sv [hunk 4]
```diff
@@ -93,16 +93,22 @@ module dm_csrs #(
...
     logic [NrHarts/2**15:0][31:0] halted_reshaped2;
     logic [(NrHarts/2**10+1)*32-1:0] halted_flat1;
     logic [(NrHarts/2**15+1)*32-1:0] halted_flat2;
     logic [32-1:0] halted_flat3;
 
     // haltsum0
-    assign halted_reshaped0 = halted_i;
-    assign haltsum0         = halted_reshaped0[hartsel_o[19:5]];
+    always_comb begin
+        halted              = '0;
```
Comment (NilsGraf) [inline src/dm_csrs.sv]: Hi @zarubaf and @bluewww,  I added this change, which was suggested by Pirmin. In our project, we use NrHarts = 1. This change is to support NrHarts = 1 and other cases < 32. Please take a look. Thanks!

**Comment by NilsGraf** [comment]
Hi @zarubaf, @bluewww, @imphil,  could you please review this PR? Thanks

**Comment by bluewww** [comment]
Sorry I havent gotten around to test it yet. I'll give it a spin later today. Do you know how much this solves for #11?

File: src/dm_csrs.sv [hunk 5]
```diff
@@ -93,16 +93,22 @@ module dm_csrs #(
     localparam dm::dm_csr_e ProgBufEnd = dm::dm_csr_e'((dm::ProgBuf0 + {4'b0, dm::ProgBufSize}));
 
     logic [31:0] haltsum0, haltsum1, haltsum2, haltsum3;
-    logic [NrHarts/2**5 :0][31:0] halted_reshaped0;
+    logic [((NrHarts-1)/2**5 + 1) * 32 - 1 : 0] halted;
+    logic [(NrHarts-1)/2**5:0][31:0] halted_reshaped0;
```
Comment (bluewww) [inline src/dm_csrs.sv]: I double checked this, it's correct also for NrHarts > 32

**Comment by NilsGraf** [comment]
Thanks for the LGTM. @zarubaf, @bluewww, @imphil, could either of you merge it? It says "base branch restricts merging to authorized users."

**Comment by zarubaf** [comment]
Sorry I don't have the bandwidth to check this right now. But @bluewww is anyway the go to person. He should also have all the permissions to eventually merge.

**Comment by NilsGraf** [comment]
Regarding #11: This PR definitely helps for the case Nrharts = 1, but not sure about all the other possible cases.

**Comment by bluewww** [comment]
I wrote a small testbench and checked a few edge cases of the haltsums. Tomorrow I'll give a last look (squeezing it through tests) and then I merge if nothing else turns up.

**Comment by NilsGraf** [comment]
SGTM, thanks!
